spec/fixtures/site_settings/locale_override.yml

```diff
@@ -0,0 +1,14 @@
+basic:
+  default_locale:
+    default: 'en'
+    enum: 'LocaleSiteSetting'
+
+search:
+  min_search_term_length:
+    client: true
+    default:
+      development: 1
```

Comment:
Yeah I see the problem cause now you can specify per env and per locale defaults, resolution order is confusing... 

I guess, 

per-env is priority 1 default
per-locale is priority 2 default
default is priority 3

codifying this in a config file is somewhat tricky, can you do a dev topic on meta with possible resolutions here, we need to sort out the "right" format here.